elastica/contact_forces.py

```diff
@@ -725,3 +843,148 @@ def apply_contact(
             self.k,
             self.nu,
         )
+
+
+class RodMeshSurfaceContactWithGridMethod(NoContact):
+    """
+    This class is for applying contact forces between rod-mesh_surface.
+    First system is always rod and second system is always mesh_surface.
+
+    Examples
+    --------
+    How to define contact between rod and mesh_surface.
+
+    >>> simulator.detect_contact_between(rod, mesh_surface).using(
+    ...    RodMeshSurfaceContactWithGridMethod,
+    ...    k=1e4,
+    ...    nu=10,
+    ...    surface_tol=1e-2,
+    ... )
+    """
+
+    def __init__(
+        self, k: float, nu: float, faces_grid: dict, grid_size: float, surface_tol=1e-4
+    ):
+        """
+
+        Parameters
+        ----------
+        k: float
+            Stiffness coefficient between the plane and the rod-like object.
+        nu: float
+            Dissipation coefficient between the plane and the rod-like object.
+        faces_grid: dict
+            Dictionary containing the grid information of the mesh surface.
+        grid_size: float
+            Grid size of the mesh surface.
+        surface_tol: float
+            Penetration tolerance between the surface and the rod-like object.
+
+        """
+        super(RodMeshSurfaceContactWithGridMethod, self).__init__()
+        # n_faces = faces.shape[-1]
+        self.k = k
+        self.nu = nu
+        self.faces_grid = faces_grid
+        self.grid_size = grid_size
+        self.surface_tol = surface_tol
+
+    def _check_systems_validity(
+        self,
+        system_one: SystemType,
+        system_two: AllowedContactType,
+        faces_grid: dict,
+        grid_size: float,
+    ) -> None:
+        """
+        This checks the contact order and type of a SystemType object and an AllowedContactType object.
+        For the RodMeshSurfaceContact class first_system should be a rod and second_system should be a mesh_surface;
+        morever, the imported grid's attributes should match imported rod-mesh_surface(in contact) grid's attributes.
+
+        Parameters
+        ----------
+        system_one
+            SystemType
+        system_two
+            AllowedContactType
+        """
+        if not issubclass(system_one.__class__, RodBase) or not issubclass(
+            system_two.__class__, MeshSurface
+        ):
+            raise TypeError(
+                "Systems provided to the contact class have incorrect order/type. \n"
+                " First system is {0} and second system is {1}. \n"
+                " First system should be a rod, second should be a mesh surface".format(
+                    system_one.__class__, system_two.__class__
+                )
+            )
+
+        elif not faces_grid["grid_size"] == grid_size:
```

Comment:
what I would do here is check if the faces_grid["grid_size"] == max(2*system_one.radius[0],system_one.rest_lengths[0]). Essentially, checking if the grid_size of faces_grid is appropriate for the given rod.